src/Directory.Build.props

```diff
@@ -1,5 +1,29 @@
 <Project ToolsVersion="15.0">
 
+  <PropertyGroup>
+    <IsMonoAndroid>false</IsMonoAndroid>
+    <IsMonoAndroid Condition="
+		$(TargetFramework.ToLower().StartsWith('monoandroid'))
+		or '$(TargetFramework)'=='net6.0-android'
+		or '$(TargetFramework)'=='net7.0-android'
+		">true</IsMonoAndroid>
+
+    <IsXamarinIOS>false</IsXamarinIOS>
+    <IsXamarinIOS Condition="
+		$(TargetFramework.ToLower().StartsWith('xamarinios'))
+		or '$(TargetFramework)'=='net6.0-ios'
+		or '$(TargetFramework)'=='net7.0-ios'
+		">true</IsXamarinIOS>
+
+    <IsXamarinMac>false</IsXamarinMac>
+    <IsXamarinMac Condition="
+		$(TargetFramework.ToLower().StartsWith('xamarinmac'))
+		or '$(TargetFramework)'=='net6.0-macos'
+		or '$(TargetFramework)'=='net7.0-macos'
+		">true</IsXamarinMac>
+  </PropertyGroup>
+
+  <Import Project="PlatformItemGroups.props" />
```

Comment:
@jeromelaban I added this import to ensure that platform file globbing works for all projects, and I moved the properties up so that they are available before they are used.

Reply:
Next step to be done in a future PR is cleaning this up a little bit. The property names are not accurate. The TFMs `net6.0-android` and `net7.0-android` are not mono, so `IsMonoAndroid` should be renamed to `IsAndroid`. I'll do that separately as this PR has already enough of changes.

Reply:
Still seems to be acting up a bit 🤔

Reply:
MSBuild ordering of things has always confused me 😕 

I'm trying now with having this Import in targets instead of props (which should be imported a bit later).

If it's not late enough, I'll revert and add the import to the individual projects that need them.

Reply:
Add this import at the top level is likely to cause a bunch of issues, but we can try :) As for IsMonoAndroid and others, this can cause issues with the import ordering, where this file may be included before any of the sub-properties (derived from TargetFramework) are being set.